README.md

```diff
@@ -45,12 +45,15 @@ void main() async {
     "calculate",
     description: 'Perform basic arithmetic operations',
     inputSchemaProperties: {
-      'operation': {
-        'type': 'string',
-        'enum': ['add', 'subtract', 'multiply', 'divide'],
+      'properties': {
+        'operation': {
+          'type': 'string',
+          'enum': ['add', 'subtract', 'multiply', 'divide'],
+        },
+        'a': {'type': 'number'},
+        'b': {'type': 'number'},
       },
-      'a': {'type': 'number'},
-      'b': {'type': 'number'},
+      'required': ['operation', 'a', 'b'],
```

Comment:
I think we should make it aligned with this https://github.com/modelcontextprotocol/modelcontextprotocol/blob/e7999bf7c268f786df89b0ffb694bdd62cc84827/schema/2025-03-26/schema.ts#L800

It means that we may want to define `InputSchemaProperty` type that has `type`, `properties`, and `required` And the `properties` is also the same type.(recursion)

Probably we could also leverage `sealed class` since it can be `string` or `object` type.